src/Game/Bank.java

```diff
@@ -18,191 +18,1056 @@
+          for (int i = 0; i < bankNumberOfItemsActual; i++) {
+            if (bankItemsActual[i] == itemId && bankItemsShown[i] && !addedToSortedBank[itemId]) {
+              tmpNewBankItems[idx] = bankItemsActual[i];
+              tmpNewBankItemsCount[idx] = bankItemCountsActual[i];
+              ++idx;
+              break;
+            }
+          }
+        }
+
+        sortedBank = true;
+      } else {
+        Client.displayMessage(
+            "@lre@You don't have any user bank sort settings defined.", Client.CHAT_QUEST);
+        Client.displayMessage(
+            "@whi@To use this button, open the settings with @mag@<ctrl-o> @whi@and then navigate to the @lre@\"Bank\"@whi@ tab.",
+            Client.CHAT_QUEST);
+        disableUserButton = true;
+      }
+    }
+
+    // still apply filter even if no sort is applied
+    if (!sortedBank) {
+      for (int i = 0; i < bankNumberOfItemsActual; i++) {
+        if (bankItemsShown[i]) {
+          tmpNewBankItems[idx] = bankItemsActual[i];
+          tmpNewBankItemsCount[idx] = bankItemCountsActual[i];
+          idx++;
+        }
+      }
+    }
+    tmpNewBankCount = idx;
+  }
+
+  private static void writeInventoryToEndOfBank() {
+    for (int inventoryIdx = 0;
+        inventoryIdx < Client.inventory_count && Client.count_items_bank < Client.bank_items_max;
+        ++inventoryIdx) {
+
+      int inventoryItemId = Client.inventory_items[inventoryIdx];
+      // Logger.Info(inventoryIdx + ". Looking for " +
+      // Item.item_name[Client.inventory_items[inventoryIdx]]);
+      boolean bankHasItem = false;
+
+      for (int bankIdx = 0; bankIdx < Client.count_items_bank; ++bankIdx) {
+        if (inventoryItemId == Client.bank_items[bankIdx]) {
+          bankHasItem = true;
+          break;
+        }
+      }
+
+      if (!bankHasItem) {
+        Client.bank_items[Client.count_items_bank] = inventoryItemId;
+        Client.bank_items_count[Client.count_items_bank] = 0;
+        ++Client.count_items_bank;
+      }
+    }
+  }
+
+  private static void writeProcessedToClient() {
+    for (int i = 0; i < Client.bank_items.length; i++) {
+      Client.bank_items[i] = tmpNewBankItems[i];
+      Client.bank_items_count[i] = tmpNewBankItemsCount[i];
+      Client.new_bank_items[i] = tmpNewBankItems[i];
+      Client.new_bank_items_count[i] = tmpNewBankItemsCount[i];
+    }
+    Client.count_items_bank = tmpNewBankCount;
+    Client.new_count_items_bank = tmpNewBankCount;
+  }
+
+  private static void calculateBankValue() {
+    int value = 0;
+    for (int idx = 0; idx < Client.new_count_items_bank; idx++) {
+      int itemId = Client.new_bank_items[idx];
+      if (itemId > 1289) {
+        Logger.Info("itemId " + itemId + " at position " + idx + " @|red fwehhhh|@");
+      } else {
+        value += BankSorters.itemValues[itemId] * Client.new_bank_items_count[idx];
+      }
+    }
+    bankValue = String.format("Bank Value: (%d gp)", value);

```

Comment:
This method is very sweet. However, that the way it operates on shown bank items for calculation, maybe should be set to display if filtered to be "Bank Value ⧩: (%d gp)" or "Filtered Bank: (%d gp)" or "Filtered Value: (%d gp)"

Reply:
Also maybe there could be a setting if it wants to be displayed as gp / k / M / B and or colorized

Reply:
I agree yes. And market value should be a tick box as well.